lib/widgets/forms/datepicker_field.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+import 'package:fokus/utils/app_locales.dart';
+
+class DatePickerField extends StatefulWidget {
+	final String labelText;
+	final String errorText;
+	final String helperText;
+	final IconData icon;
+	final Function dateSetter;
+	final TextEditingController dateController;
+	final bool canBeEmpty;
+	final DateTime rangeFromDate;
+	final DateTime rangeToDate;
+	final DateTime initialDate;
+	final Function callback;
+
+	DatePickerField({
+		@required this.labelText,
+		@required this.dateController,
+		@required this.dateSetter,
+		@required this.callback,
+		this.errorText,
+		this.helperText,
+		this.icon = Icons.event,
+		this.canBeEmpty = true,
+		this.rangeFromDate,
+		this.rangeToDate,
+		this.initialDate
+	});
+
+	@override
+  State<StatefulWidget> createState() => new _DatePickerFieldState();
+
+}
+
+class _DatePickerFieldState extends State<DatePickerField> {
+	bool _showClearButton = false;
+
+	Future<DateTime> _selectDate(BuildContext context) async {
+		DateTime initial = (widget.dateController.text != '' ? 
+			DateTime.tryParse(widget.dateController.text) :
+			(widget.rangeFromDate != null ? 
+				widget.rangeFromDate : 
+				(widget.rangeToDate != null ? 
+					widget.rangeToDate :
+					DateTime.now()
+				)
+			)
```

Comment:
This is a new record for ternary operator nesting! But srsly trying to understand what it does hurts :P

Reply:
It's for deciding what is the initial date for date picker widget. It has a lot of conditionals (it's a chain of: set this unless <condition>...). I will reformat this to classic ifs.